From: Joan Torres Lopez <joantolo@redhat.com>
Date: Wed, 12 Aug 2026 17:08:36 +0200
Subject: gdm/userVerifier: Use every() in needsUsername()
MIME-Version: 1.0
Content-Type: text/plain; charset="utf-8"
Content-Transfer-Encoding: 8bit

needsUsername() determines whether to show a username prompt on reset.
If at least one service can operate without a username, we should skip the
username prompt — so the result should be true only when *every* service
requires a username. Using some() incorrectly returned true when
at least one service needed a username, even if another didn't.

This is to fix credentialManagers authentication which automatically selects
the username and authenticates.

Part-of: <https://gitlab.gnome.org/GNOME/gnome-shell/-/merge_requests/4338>

Origin: https://gitlab.gnome.org/GNOME/gnome-shell/-/commit/66d338b
---
 js/gdm/userVerifier.js | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/js/gdm/userVerifier.js b/js/gdm/userVerifier.js
index 7428450..da3965e 100644
--- a/js/gdm/userVerifier.js
+++ b/js/gdm/userVerifier.js
@@ -112,7 +112,7 @@ export class ShellUserVerifier extends Signals.EventEmitter {
     }
 
     needsUsername() {
-        return this._authServices.some(s => s.needsUsername());
+        return this._authServices.every(s => s.needsUsername());
     }
 
     setDriverService(serviceName) {
